Platform/AlderlakeBoardPkg/BoardConfigAdlN.py

```diff
@@ -0,0 +1,516 @@
+
+        self.LOADER_RSVD_MEM_SIZE = 0x500000
+
+        # If mulitple VBT table support is required, list them as:
+        #   {VbtImageId1 : VbtFileName1, VbtImageId2 : VbtFileName2, ...}
+        # VbtImageId is ID to identify a VBT image. It is a UINT32 number to match
+        #   the ImageId field in the VBT container.
+        # VbtFileName is the VBT file name. It needs to be located under platform
+        #   VbtBin folder.
+        self._MULTI_VBT_FILE      = {1:'VbtAdlN.dat'}
+
+        self.CFG_DATABASE_SIZE    = self.CFGDATA_SIZE
+        self._generated_cfg_file_prefix = 'Autogen_'
+
+        self._CFGDATA_INT_FILE = []
+        self._CFGDATA_EXT_FILE = [self._generated_cfg_file_prefix + 'CfgDataInt_Adln_Crb_Ddr5.dlt', self._generated_cfg_file_prefix + 'CfgDataInt_Adln_Rvp_Lpddr5.dlt']
+
+    def PlatformBuildHook (self, build, phase):
+        if phase == 'pre-build:before':
+            # create build folder if not exist
+            if not os.path.exists(build._fv_dir):
+                os.makedirs(build._fv_dir)
+
+            # Generate the dlt files based on feature
+            brd_cfg_src_dir = os.path.join(os.environ['PLT_SOURCE'], 'Platform', self.BOARD_PKG_NAME, 'CfgData')
+            brd_cfg2_src_dir = '.'
+            if hasattr(self, 'BOARD_PKG_NAME_OVERRIDE'):
+                brd_cfg2_src_dir = os.path.join(os.environ['PLT_SOURCE'], 'Platform', self.BOARD_PKG_NAME_OVERRIDE, 'CfgData')
+            for dlt_file in self._CFGDATA_EXT_FILE:
+                cfg_dlt_file  = os.path.join(brd_cfg_src_dir, dlt_file[len (self._generated_cfg_file_prefix):])
+                if not os.path.exists(cfg_dlt_file):
+                    cfg_dlt_file = os.path.join(brd_cfg2_src_dir, dlt_file[len (self._generated_cfg_file_prefix):])
+                lines         = open (cfg_dlt_file).read()
+
+                # Enable TCC in dlt file
+                if self.ENABLE_TCC:
+                    if os.path.exists(os.path.join(brd_cfg_src_dir, 'CfgData_Adln_Tcc_Feature.dlt')):
+                        lines += open (os.path.join(brd_cfg_src_dir, 'CfgData_Adln_Tcc_Feature.dlt')).read()
+                    elif os.path.exists(os.path.join(brd_cfg_src_dir, 'CfgData_Tcc_Feature.dlt')):
+                        lines += open (os.path.join(brd_cfg_src_dir, 'CfgData_Tcc_Feature.dlt')).read()
+                    elif os.path.exists(os.path.join(brd_cfg2_src_dir, 'CfgData_Tcc_Feature.dlt')):
+                        lines += open (os.path.join(brd_cfg2_src_dir, 'CfgData_Tcc_Feature.dlt')).read()
+                    else:
+                        tcc_cfg_dir = os.path.join(os.getenv('SBL_SOURCE', ''), 'Platform', self.BOARD_PKG_NAME, 'CfgData')
+                        lines += open (os.path.join(tcc_cfg_dir, 'CfgData_Tcc_Feature.dlt')).read()
+
+                # Enable TSN in dlt file
+                if self.ENABLE_TSN:
+                    if os.path.exists(os.path.join(brd_cfg_src_dir, 'CfgData_Tsn_Feature.dlt')):
```

Comment:
File name should be Adln specific one. Please change.

Reply:
thanks, resolved.